src/Ledger/Utxow.lagda

```diff
@@ -83,11 +91,18 @@ data _⊢_⇀⦇_,UTXOW⦈_ where
         open UTxOState s
         witsKeyHashes     = mapˢ hash (dom vkSigs)
         witsScriptHashes  = mapˢ hash scripts
+        inputHashes       = getInputHashes tx utxo
+        refScriptHashes   = mapˢ hash (refScripts tx utxo)
+        neededHashes      = scriptsNeeded utxo txb
+        txdatsHashes      = dom (txdats .proj₁)
+        allOutHashes      = getDataHashes (range txouts)
     in
     ∙  ∀[ (vk , σ) ∈ vkSigs ] isSigned vk (txidBytes txid) σ
-    ∙  ∀[ s ∈ scriptsP1 ] validP1Script witsKeyHashes txvldt s
+    ∙  ∀[ s ∈ mapPartial isInj₁ (txscripts tx utxo) ∩ scriptsP1 ] validP1Script witsKeyHashes txvldt s
```

Comment:
```suggestion
    ∙  ∀[ s ∈ mapPartial isInj₁ (txscripts tx utxo) ] validP1Script witsKeyHashes txvldt s
```
`scriptsP1` are only from the witnesses, so doing the intersection here removes all the scripts from references.